shared/interfaces.go

```diff
@@ -0,0 +1,13 @@
+package shared
```

Comment:
> I agree that "shared" could become a dumping ground. However, I disagree that this is for deps.
> 
> Interfaces (and currently a TChannel util function) in the shared package are for use by the ringpop, forward, replica > and swim packages. These are all separate packages that are inside the ringpop repository.

That's my point. The TChannel interface you've defined is an interface standing in as an abstraction built around a 3rd party dependency of Ringpop (including all of Ringpop's packages). `interfaces.go` is too general. I think there is a difference in these interfaces (for TChannel et. al) and interfaces that Ringpop may define and share internally. That's why I'd make the distinction between them.

> There's another reason they are in a separate package and not just in ringpop: circular dependencies...

:+1:  Good to know